Source/Editor/Foundation/InspectorTab/ModelInspector.cpp

```diff
@@ -50,14 +51,70 @@ SharedPtr<ResourceInspectorWidget> ModelInspector::MakeInspectorWidget(const Res
     return MakeShared<ModelInspectorWidget>(context_, resources);
 }
 
+class ModelInspectorSceneWidget : public SceneWidget
+{
+public:
+    ModelInspectorSceneWidget(Context* context)
+        : SceneWidget(context)
+    {
+    }
+
+    ~ModelInspectorSceneWidget() {}
+
+    void SetModel(Model* model)
+    {
+        model_ = model;
+    }
+
+    void RenderContent() override
+    {
+        if (model_)
+        {
+            auto& vertexBuffers = model_->GetVertexBuffers();
+            if (!vertexBuffers.empty())
+            {
+                ui::NewLine();
+            }
+
+            for (unsigned vertexBufferIndex = 0; vertexBufferIndex < vertexBuffers.size(); ++vertexBufferIndex)
+            {
+                const auto& vb = vertexBuffers[vertexBufferIndex];
+                const float kb = (vb->GetVertexCount() * vb->GetVertexSize()) / 1000.0f;
+                ui::Text("VertexBuffer[%u]: %u vertices (%.1f KB)", vertexBufferIndex, vb->GetVertexCount(), kb);
+
+                const float indent = 5;
+                ui::Indent(indent);
+
+                for (auto& ve : vb->GetElements())
+                {
+                    ui::Text("%s (%s)", vertexElementSemanticNames[ve.semantic_], vertexElementTypeNames[ve.type_]);
+                }
+
+                ui::Indent(-indent);
+            }
+
+            if (!vertexBuffers.empty())
+            {
+                ui::NewLine();
+            }
+        }
+
+        SceneWidget::RenderContent();
```

Comment:
`TreeNode()` with info should be moved after this call, so we still have a preview first thing we see as it is the most important.

Reply:
Done

![image](https://github.com/user-attachments/assets/f7398cd1-5fee-428a-ad48-6dd5fe665bf1)